capstone-app/src/app/modules/maps/map/map.component.spec.ts

```diff
@@ -1,20 +1,82 @@
+/// <reference types="googlemaps" />
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { MapComponent } from './map.component';
+import { GoogleMapsModule } from '@angular/google-maps';
+import { By } from '@angular/platform-browser';
+
 
 describe('MapComponent', () => {
   let component: MapComponent;
   let fixture: ComponentFixture<MapComponent>;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ MapComponent ]
+      declarations: [ MapComponent ],
+      imports: [ GoogleMapsModule ]
     })
     .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(MapComponent);
-    component = fixture.componentInstance;
     fixture.detectChanges();
   });
+
+  it('should create the app', () => {
+    // const fixture = TestBed.createComponent(MapComponent);
+    // const app = fixture.componentInstance;
+    expect(fixture.componentInstance).toBeTruthy();
+  });
+
+  it('test map marker click event', () => {
+    const fixture = TestBed.createComponent(MapComponent);
+    fixture.detectChanges();
+
+    const coordinates = new google.maps.LatLng({lat: 26.011760, lng:  -80.139050});
+
+    fixture.componentInstance.markers.push(coordinates);
+    fixture.detectChanges();
```

Comment:
Is it possible to trigger an emission from mapRightclick on the google-map instead. Something likr:
const googleMapDebug = fixture.debugElement.query(By.css('google-map'));
const googleMapDebugComponent = googleMapDebug.componentInstance;

// trigger an emission of a google.maps.MouseEvent object from the google-map component
// https://developers.google.com/maps/documentation/javascript/reference/map#MouseEvent
googleMapDebugComponent.mapRightclick.emit({latLng: coordinates});

If not, then is it possible to pass a google.maps.MouseEvent object to the addMarker method?